test/e2e/data_setup/schema/recordset/product-recordset.json

```diff
@@ -825,7 +869,165 @@
           "display": "Categories"
         },
         "tag:isrd.isi.edu,2016:visible-columns": {
-          "*": ["id", "term"]
+          "*": ["id", "term", "type"]
+        },
+        "tag:isrd.isi.edu,2016:visible-foreign-keys" : {
+          "*" : [
+              ["product-recordset", "fk_facilities"]
+          ]
+        }
+      }
+    },
+    "facilities": {
+      "comment": "Type of facilities ('Luxury/Upscale/Basic')",
+      "kind": "table",
+      "keys": [
+        {
+          "comment": "vocabulary",
+          "annotations": {},
+          "unique_columns": [
+            "id"
+          ]
+        },
+        {
+          "comment": "vocabulary",
+          "annotations": {},
+          "unique_columns": [
+            "type"
+          ]
+        }
+      ],
+      "entityCount": 0,
+      "foreign_keys": [],
+      "table_name": "facilities",
+      "schema_name": "product-recordset",
+      "column_definitions": [
+        {
+          "comment": null,
+          "name": "id",
+          "default": null,
```

Comment:
This won't make a difference in the tests because the data you added always has ids, but this line conflicts with the next one. Having `nullok: false` means that the value **CANNOT** be null, but this line says to set it to `null` if it isn't present. It creates a confusing template.

If we were to look at this table in `recordedit` with both being defined, the UX will always enforce the `nullok: false` as a required field so data (for this column) won't ever be able to be submitted with `id = null`.